management-cluster-access/README.md

```diff
@@ -0,0 +1,95 @@
+# Ensure no single point of failure in management cluster access
+
+## Access to MCs for giant swarm staff in vintage vs the CAPI product
+
+Our default access method to management clusters is now single-sign-on.
+For Giant Swarm staff this is done via `dex` using github as the _only_ identity provider. 
+
+Our vintage product also offers the possibility to create a client certificate to access the management cluster.
+This means that in the event of github or `dex` being unavailable, we do not get locked out of the management cluster.
+
+For pure CAPI installations on the other hand, no other authentication method is available at this point.
+This poses a real risk to get locked out of the management cluster in case of problems with OIDC.
+We also rely on the github api being available as an external dependency we can not control.
+
+Furthermore, loss of access to the management clusters implies loss of access to all workload clusters that do not have OIDC access for giant swarm staff set up.
+This is due to the fact that in order to create a client certificate for these workload clusters, we need to access the management cluster first.
+At this moment this likely applies to all workload clusters on pure CAPI installations.
+
+However, even if OIDC access was set up, it would still leave github as a single point of failure.
+
+Let's ensure that management cluster access is highly available for operations through GS staff in the future.
+
+## References
+
+- [PKI epic](https://github.com/giantswarm/giantswarm/issues/15981)
+- [Issue regarding client cert creation for CAPO ](https://github.com/giantswarm/giantswarm/issues/21740)
+- [story for backup idp in dex](https://github.com/giantswarm/roadmap/issues/603)
+- [proposal to use azure AD for GS staff](https://github.com/giantswarm/giantswarm/issues/21627)
+- 
+
+## Possible solutions
+
+### Client certificates
+
+In the vintage product, client certificates can be created from outside the cluster using `vault` for signing.
+At the moment this is not the case for the CAPI product.
+
+- This method of authentication is completely separate from SSO and does not have shared dependencies.
+- As of now, client certificate creation is harder to monitor and control than SSO.
+- Client certificates should be short lived and therefore would need to be recreated by users frequently.
+- A long living client-certificate granting admin access is a security risk since permissions can not be revoked.
```

Comment:
We can limit TTL duration in our API

Reply:
Agreed. IMO that has to be a requirement if we want to support this.